src/components/NavBar/index.tsx

```diff
@@ -0,0 +1,125 @@
+'use client';
+
+import { useMemo } from 'react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { usePathname } from 'next/navigation';
+import COLORS from '@/styles/colors';
+import { Flex } from '@/styles/containers';
+import { useAuth } from '@/utils/AuthProvider';
+import { useProfile } from '@/utils/ProfileProvider';
+import IJPlogo from '~/public/images/ijp-logo.webp';
+import { ProfileButton, SmallLinkButton } from '../Buttons';
+import * as Styles from './style';
+
+export default function NavBar() {
+  const profile = useProfile();
+  if (!profile) throw new Error('Profile must be defined.');
+
+  const auth = useAuth();
+  if (!auth) throw new Error('Auth Must be defined.');
+
+  const AuthButtonView = useMemo(() => {
+    if (profile.profileReady && auth.userId)
+      return (
+        <ProfileButton href="/settings">
+          {profile.profileData?.first_name || 'Profile'}
+        </ProfileButton>
+      );
+
+    return (
+      <>
+        <SmallLinkButton
+          $primaryColor={COLORS.blueMid}
+          $secondaryColor={COLORS.blueDark}
+          href="/login"
+        >
+          Log In
+        </SmallLinkButton>
+        <SmallLinkButton
+          $primaryColor="white"
+          $secondaryColor={COLORS.blueDark}
+          href="/signup"
+          style={{ color: COLORS.blueMid }}
+        >
+          Sign Up
+        </SmallLinkButton>
+      </>
+    );
+  }, [profile]);
+
+  function useActiveStatus(path: string): boolean {
+    const currentPath = usePathname();
+    return currentPath.includes(path);
+  }
+
+  type NavLink = {
+    name: string;
+    path: string;
+    active: boolean;
+  };
+  const navlinks: NavLink[] = [
+    { name: 'Cases', path: '/cases', active: useActiveStatus('/cases') },
+    {
+      name: 'Limited Case Assignments',
+      path: '/limited-case-assignments',
+      active: useActiveStatus('/limited-case-assignments'),
+    },
+    {
+      name: 'Language Support',
+      path: '/language-support',
+      active: useActiveStatus('/language-support'),
+    },
+  ];
+
```

Comment:
for static data like this, let's move it outside the function so it doesn't get re-computed when the navbar re-renders

note: every time a component re-renders, everything inside the component definition (except those wrapped in memo functions, for example) is re-run! so putting static data like this here would be pretty inefficient

Reply:
We can move the type definition outside of the function, but we get this lint error when moving `navlinks`:
 
<img width="811" alt="image" src="https://github.com/calblueprint/immigration-justice-project/assets/84427964/d9c157e5-5832-4a0c-8cea-f313dbc0ab68">

Reply:
ah, i think it's better not to use hooks more than once if we can help it.
here, it may be better to just get the pathname using `usePathname` and here, instead of the hook, make an anonymous arrow function like this:
```ts
(pathname: string) => pathname === '/language-support'
```
then call that function in the rendering function, passing in the current pathname

Reply:
Is there an issue with calling the hook multiple times? I feel like using the hook may be better for readability.

Reply:
Primary issue is redundancy. I don't know how `usePathname` impacts performance, but it feels wasteful to call it multiple times when we can just call it once and store it. Further, doing it the other way allows us to store this data outside of the component function, so we can avoid re-defining it each time this component re-renders.

This is ultimately a nit, and I agree that this may offer slight performance gains. However, I don't think it sacrifices much in terms of readability (we just turn the status from a variable into a function).

Reply:
Yeah, I agree that it's best to not call `usePathname` multiple times if we can avoid it. Thoughts on this approach instead of creating a function:


``` 
  const currentPath = usePathname();

...
...
...

const navlinks: NavLink[] = [
    { name: 'Cases', path: '/cases', active: currentPath === '/cases' },
    {
      name: 'Limited Case Assignments',
      path: '/limited-case-assignments',
      active: currentPath === '/limited-case-assignments',
    },
    {
      name: 'Language Support',
      path: '/language-support',
      active: currentPath === '/language-support',
    },
  ];```

Reply:
I noticed a pattern of the `path` being the same as the comparison. Could we just add a check like `currentPath = link.path` in the rendering function? Then we can move the nav links definition outside of the component function

Reply:
Wait I don't know how I didn't see that before!!